File: frontend/src/lib/sessionHubEvents.ts
export const SessionHubEvent = {
  UserJoinedSession: 'UserJoinedSession',
  UserLeftSession: 'UserLeftSession',
  MovieSuggested: 'MovieSuggested',
  ReadyToVote: 'ReadyToVote',
  VotingStarted: 'VotingStarted',
  VoteSubmitted: 'VoteSubmitted',
  VotingCompleted: 'VotingCompleted',
  MovieSelected: 'MovieSelected',
  TurnPassed: 'TurnPassed',
  MovieWatched: 'MovieWatched',
  SessionError: 'SessionError',
} as const

export interface SessionErrorPayload {
  sessionId: string
  code: string
  message: string
}

File: src/CozyCinema.Application/RealTime/SessionHubEvents.cs
namespace CozyCinema.Application.RealTime;

public static class SessionHubEvents
{
    public const string UserJoinedSession = nameof(UserJoinedSession);
    public const string UserLeftSession = nameof(UserLeftSession);
    public const string MovieSuggested = nameof(MovieSuggested);
    public const string ReadyToVote = nameof(ReadyToVote);
    public const string VotingStarted = nameof(VotingStarted);
    public const string VoteSubmitted = nameof(VoteSubmitted);
    public const string VotingCompleted = nameof(VotingCompleted);
    public const string MovieSelected = nameof(MovieSelected);
    public const string TurnPassed = nameof(TurnPassed);
    public const string MovieWatched = nameof(MovieWatched);
    public const string SessionError = nameof(SessionError);

    public static string GroupName(Guid sessionId) => $"session:{sessionId}";
}
